### PR TITLE
Dialogs Implementation with Info and Agreement Types

### PR DESCRIPTION
**Pull request by bilgehankalkan**
Dialog UI components.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Dialogs is a BottomSheetDialog variants.

## Screenshots
* Info Dialog:

* Agreement Dialog:
  

## Motivation and Context
To present user a modern and user friendly dialogs.

## How Has This Been Tested?
Tested on Android 7.1 device.

## Checklist:
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BaseBottomSheetDialog.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.trendyol.dialog.R
+
+abstract class BaseBottomSheetDialog<DB : ViewDataBinding> : BottomSheetDialogFragment() {
```
Comment (bsobe) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BaseBottomSheetDialog.kt]: Dialog and DialogFragment are different types. That's why I think that class name should be renamed with **BaseBottomSheetDialogFragment**

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BindingAdapters.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.trendyol.uicomponents.dialogs
+
+import android.view.View
+import androidx.annotation.DrawableRes
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.databinding.BindingAdapter
+
+@BindingAdapter("drawable")
+internal fun AppCompatImageView.setDrawable(@DrawableRes drawableResId: Int) {
```
Comment (bsobe) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BindingAdapters.kt]: func name should be rename with setDrawableResource or any similar name

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        binding.executePendingBindings()
+    }
+
+    fun showDialog(fragmentManager: FragmentManager) {
+        show(fragmentManager, TAG)
+    }
+
+    companion object {
+
+        const val TAG: String = "TRENDYOL_BOTTOM_SHEET_DIALOG"
```
Comment (bsobe) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt]: Are you sure to use that naming (TRENDYOL) on open source lib?

File: libraries/dialogs/src/main/res/layout/dialog_info.xml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                app:layout_constraintEnd_toStartOf="@id/imageClose"
+                app:layout_constraintStart_toStartOf="@id/guideStart"
+                app:layout_constraintTop_toTopOf="@id/viewTitleBackground"
+                tools:text="Lorem Ipsum" />
+
+            <androidx.appcompat.widget.AppCompatImageView
+                android:id="@+id/imageClose"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:padding="8dp"
```
Comment (bsobe) [inline libraries/dialogs/src/main/res/layout/dialog_info.xml]: hardcoded values may change with using dimens.xml

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import android.text.SpannableString
+import android.text.Spanned
+import androidx.annotation.DrawableRes
+import androidx.fragment.app.FragmentManager
+import com.trendyol.dialog.R
+import com.trendyol.dialog.databinding.DialogInfoBinding
+
+class Dialog internal constructor(
+    private val title: String? = null,
+    private val showCloseButton: Boolean? = null,
```
Comment (bsobe) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt]: showCloseButton never used. why?

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt (hunk 3 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt]: Why not? This library is under "Trendyol Open Source" organization.

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BaseBottomSheetDialog.kt (hunk 1 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BaseBottomSheetDialog.kt]: Done.

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BindingAdapters.kt (hunk 2 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/BindingAdapters.kt]: Done.

File: libraries/dialogs/src/main/res/layout/dialog_info.xml (hunk 4 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/res/layout/dialog_info.xml]: Done.

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt (hunk 5 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Dialog.kt]: Missed, implemented.

**Pull request merged**